harness/determined/common/experimental/experiment.py

```diff
@@ -52,17 +52,32 @@ def __init__(
         self._id = experiment_id
         self._session = session
 
+        # These properties may be mutable and will be set by _hydrate()
+        self.config = None
+
     @property
     def id(self) -> int:
         return self._id
 
-    def _get(self) -> bindings.v1Experiment:
+    def _get(self, session: api.Session = None) -> bindings.v1Experiment:
         """
         _get fetches the main GET experiment endpoint and parses the response.
         """
-        resp = bindings.get_GetExperiment(self._session, experimentId=self._id)
+        if session is None:
+            session = self._session
+        resp = bindings.get_GetExperiment(session, experimentId=self.id)
         return resp.experiment
 
+    def _hydrate(self, exp: bindings.v1Experiment):
```

Comment:
I really like how explicit the design is here with what is/isn't mutable via hydration.